test/helpers/kubectl.go

```diff
@@ -860,12 +860,30 @@ func (kub *Kubectl) WaitForKubeDNSEntry(serviceName, serviceNamespace string) er
 		// an IP of the pod itself, not ClusterIPNone, which is what Kubernetes
 		// shows as the IP for the service for headless services.
 		if serviceIP == v1.ClusterIPNone {
-			res := kub.ExecShort(fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
-			_ = kub.ExecShort(fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
+			ress, err := kub.ExecInPods(context.TODO(), "kube-system", "k8s-app=cilium-test-logs", fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP), true)
+			if err != nil {
+				logger.Debugf("failed to run dig in log-gatherer pods")
+				return false
+			}
+			var res *CmdRes
+			for _, v := range ress {
+				res = v
+				break
+			}
+			//_ = kub.ExecShort(fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
 			return res.WasSuccessful()
 		}
 		log.Debugf("service is not headless; checking whether IP retrieved from DNS matches the IP for the service stored in Kubernetes")
-		res := kub.ExecShort(fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
+		ress, err := kub.ExecInPods(context.TODO(), "kube-system", "k8s-app=cilium-test-logs", fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP), true)
+		if err != nil {
+			logger.Debugf("failed to run dig in log-gatherer pods")
+			return false
+		}
+		var res *CmdRes
+		for _, v := range ress {
```

Comment:
What is the rationale behind this loop? Is it because we only need one of the results at all?

Reply:
Changed it to different helper function which I created after this code.